lib/fog/compute/google/requests/insert_server.rb

```diff
@@ -133,8 +133,21 @@ def handle_networks(options)
             network = networks.get(network)
           end
 
+          network_interface = { "network" => network.get_self_link_attr }
+
+          if options.key? "subnetwork"
+            subnetwork = options.delete "subnetwork"
+            # Objectify the subnetwork if needed
+            unless subnetwork.is_a? Subnetwork
+              # Extract region_name from zone_name	
+              l = zone_name.split(/-/)
+              region_name = l[0] + "-" + l[1]
```

Comment:
Avoid getting options by splitting parameters based on delimiters. Those can change and string manipulation is finicky.